dotCMS/src/main/java/com/dotcms/saml/DotIdentityProviderConfigurationImpl.java

```diff
@@ -0,0 +1,126 @@
+package com.dotcms.saml;
+
+import com.dotcms.security.apps.AppSecrets;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotcms.security.apps.Secret;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import io.vavr.control.Try;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * Default implementation to retrieve the configuration from apps
+ * @author jsanca
+ */
+public class DotIdentityProviderConfigurationImpl implements IdentityProviderConfiguration {
+
+    private final AppsAPI appsAPI;
+    private final Host    host;
+
+    public DotIdentityProviderConfigurationImpl(final AppsAPI appsAPI, final Host host) {
+
+        this.appsAPI = appsAPI;
+        this.host    = host;
+
+    }
+
+    private final Optional<Secret> findSecret (final String key) {
```

Comment:
Codacy found an issue: [Unnecessary modifier 'final' on method 'findSecret': private methods cannot be overridden](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)